mr/src/main/java/org/apache/iceberg/mr/InputFormatConfig.java

```diff
@@ -203,6 +202,44 @@ public static Schema readSchema(Configuration conf) {
     return readColumns != null && readColumns.length > 0 ? readColumns : null;
   }
 
+  /**
+   * Get Hadoop config key of catalog type based on catalog name and {@link #CATALOG_TYPE_TEMPLATE}
+   * @param catalogName catalog name
+   * @return Hadoop config key of catalog type for the catalog name
+   */
+  public static String catalogTypeConfigKey(String catalogName) {
+    return String.format(InputFormatConfig.CATALOG_TYPE_TEMPLATE, catalogName);
```

Comment:
shall we add `type` to the `CatalogProperties` as well so we can reuse `catalogPropertyConfigKey` here?

Reply:
It was not added because `type` is not consistent across engines. Flink has to use `catalog-type` instead of `type` because `type` is a reserved config key.

With that being said, we can probably add it and specify that Flink has to use `catalog-type`. Let me do that.

Reply:
Yeah that's actually one way we can consider, @rdblue @aokolnychyi , would it be acceptable if we unify all engines to ause `catalog-type` instead, and `type` is just an alias for `catalog-type` in Spark and Hive?

Reply:
So while Ryan and Anton have not replied, I will keep it like this, we can always add `catalog-type` as a new catalog properties in a new PR.

Reply:
In addition, I am treating `CatalogUtil.ICEBERG_CATALOG_TYPE` as the catalog property to use, and removed reference to `InputFormatConfig.CATALOG_TYPE_TEMPLATE`. I also added documentation inside `CatalogUtil` to make things clear.

I don't personally like the fact that `CatalogUtil.ICEBERG_CATALOG_TYPE` is in `CatalogUtil` instead of `CatalogProperties`, but we already released it in 0.11 as public variables so I decide to just keep it this way.